### ISSUE TITLE
gometric/opcache-dashboard:0.5.1 docker image have problem with "invalid memory address or nil pointer dereference"

### ISSUE THREAD

**Issue by allanian** [GoMetric/opcache-dashboard#22]
Thank you man for that dashboard with prometheus metrics for opcache and apcu in one place, its a rly helpful tool. Its working perfectly on 0.5.0, but on 0.5.1 generate that issue:

### docker-compose.yml part
```
    opcache-dashboard:
    container_name: opcache-dashboard
    image: gometric/opcache-dashboard:0.5.1
    restart: unless-stopped
    command: "opcache-dashboard -verbose -config config.yaml"
    ports:
      - "42042:42042"
    expose:
      - "42042"
    volumes:
      - ./config/opcache-dashboard/config.yaml:/config.yaml:ro
    networks:
      - prom-stack`
```

### tracelog
```
2021/11/11 20:31:13 Reading configuration from '/config.yaml'
2021/11/11 20:31:13 Starting observer with refresh interval 5 seconds
2021/11/11 20:31:13 Starting HTTP server at 0.0.0.0:42042
2021/11/11 20:31:13 Observing http://rv-site-back04.company.ru/public/agent-pull.php?scripts=1
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x87a692]

goroutine 42 [running]:
github.com/GoMetric/opcache-dashboard/metrics.(*PrometheusMetricSender).Send(0xc00020e088, 0xc00021a1a8, 0x8, 0xc00021a1c6, 0x7, 0xc000220240, 0x1d, 0xc0003929c0, 0xc0003a2150, 0x6, ...)
	/opcache-dashboard/metrics/prometheus_collector.go:69 +0x492
github.com/GoMetric/opcache-dashboard/observer.(*Observer).pullAgent(0xc000258000, 0xc0002240c0, 0x23, 0xc000218450, 0x1, 0x1, 0x0, 0xc00021a1a8, 0x8, 0xc00021a1c6, ...)
	/opcache-dashboard/observer/observer.go:165 +0x415
github.com/GoMetric/opcache-dashboard/observer.(*Observer).PullAgents(0xc000258000)
	/opcache-dashboard/observer/observer.go:126 +0x12d
github.com/GoMetric/opcache-dashboard/observer.(*Observer).pullAgentsOnTick(0xc000258000)
	/opcache-dashboard/observer/observer.go:117 +0x2b
created by github.com/GoMetric/opcache-dashboard/observer.(*Observer).StartPulling
	/opcache-dashboard/observer/observer.go:70 +0x6f2
2021/11/11 20:31:14 Reading configuration from '/config.yaml'
2021/11/11 20:31:14 Starting observer with refresh interval 5 seconds
2021/11/11 20:31:14 Starting HTTP server at 0.0.0.0:42042
2021/11/11 20:31:14 Observing http://rv-site-back04.company.ru/public/agent-pull.php?scripts=1
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x87a692]

goroutine 26 [running]:
github.com/GoMetric/opcache-dashboard/metrics.(*PrometheusMetricSender).Send(0xc0000a4090, 0xc0000b01a8, 0x8, 0xc0000b01c6, 0x7, 0xc0000b6240, 0x1d, 0xc000302bd0, 0xc000300cf0, 0x6, ...)
	/opcache-dashboard/metrics/prometheus_collector.go:69 +0x492
github.com/GoMetric/opcache-dashboard/observer.(*Observer).pullAgent(0xc0000ea000, 0xc0000ba0c0, 0x23, 0xc0000ae450, 0x1, 0x1, 0x0, 0xc0000b01a8, 0x8, 0xc0000b01c6, ...)
	/opcache-dashboard/observer/observer.go:165 +0x415
github.com/GoMetric/opcache-dashboard/observer.(*Observer).PullAgents(0xc0000ea000)
	/opcache-dashboard/observer/observer.go:126 +0x12d
github.com/GoMetric/opcache-dashboard/observer.(*Observer).pullAgentsOnTick(0xc0000ea000)
	/opcache-dashboard/observer/observer.go:117 +0x2b
created by github.com/GoMetric/opcache-dashboard/observer.(*Observer).StartPulling
	/opcache-dashboard/observer/observer.go:70 +0x6f2
2021/11/11 20:31:15 Reading configuration from '/config.yaml'
2021/11/11 20:31:15 Starting observer with refresh interval 5 seconds
2021/11/11 20:31:15 Starting HTTP server at 0.0.0.0:42042
2021/11/11 20:31:15 Observing http://rv-site-back04.company.ru/public/agent-pull.php?scripts=1
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x87a692]

goroutine 27 [running]:
github.com/GoMetric/opcache-dashboard/metrics.(*PrometheusMetricSender).Send(0xc00009e098, 0xc0000c01a8, 0x8, 0xc0000c01c6, 0x7, 0xc0000a2540, 0x1d, 0xc000210e40, 0xc00020ed20, 0x6, ...)
	/opcache-dashboard/metrics/prometheus_collector.go:69 +0x492
github.com/GoMetric/opcache-dashboard/observer.(*Observer).pullAgent(0xc0000b0050, 0xc0000c80c0, 0x23, 0xc00009c640, 0x1, 0x1, 0x0, 0xc0000c01a8, 0x8, 0xc0000c01c6, ...)
	/opcache-dashboard/observer/observer.go:165 +0x415
github.com/GoMetric/opcache-dashboard/observer.(*Observer).PullAgents(0xc0000b0050)
	/opcache-dashboard/observer/observer.go:126 +0x12d
github.com/GoMetric/opcache-dashboard/observer.(*Observer).pullAgentsOnTick(0xc0000b0050)
	/opcache-dashboard/observer/observer.go:117 +0x2b
created by github.com/GoMetric/opcache-dashboard/observer.(*Observer).StartPulling
	/opcache-dashboard/observer/observer.go:70 +0x6f2
2021/11/11 20:31:17 Reading configuration from '/config.yaml'
2021/11/11 20:31:17 Starting observer with refresh interval 5 seconds
2021/11/11 20:31:17 Starting HTTP server at 0.0.0.0:42042
2021/11/11 20:31:17 Observing http://rv-site-back04.company.ru/public/agent-pull.php?scripts=1
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x87a692]

goroutine 17 [running]:
github.com/GoMetric/opcache-dashboard/metrics.(*PrometheusMetricSender).Send(0xc00000e2d8, 0xc000027648, 0x8, 0xc000027666, 0x7, 0xc000025560, 0x1d, 0xc000152db0, 0xc000147620, 0x6, ...)
	/opcache-dashboard/metrics/prometheus_collector.go:69 +0x492
github.com/GoMetric/opcache-dashboard/observer.(*Observer).pullAgent(0xc000062b40, 0xc000070150, 0x23, 0xc00004deb0, 0x1, 0x1, 0x0, 0xc000027648, 0x8, 0xc000027666, ...)
	/opcache-dashboard/observer/observer.go:165 +0x415
github.com/GoMetric/opcache-dashboard/observer.(*Observer).PullAgents(0xc000062b40)
	/opcache-dashboard/observer/observer.go:126 +0x12d
github.com/GoMetric/opcache-dashboard/observer.(*Observer).pullAgentsOnTick(0xc000062b40)
	/opcache-dashboard/observer/observer.go:117 +0x2b
created by github.com/GoMetric/opcache-dashboard/observer.(*Observer).StartPulling
	/opcache-dashboard/observer/observer.go:70 +0x6f2
```

**Comment by sokil** [comment]
In 0.5.1 there is only one commit that changes code: https://github.com/GoMetric/opcache-dashboard/commit/507b6936622fe25c67a9b3ad08e967fcada749bc

In metrics/prometheus_collector.go at line 69 was added

```
"apcu_memory_free_bytes": nodeApcuStatus.SmaInfo.AvailMem,
```

This deals with your trace:

```
github.com/GoMetric/opcache-dashboard/metrics.(*PrometheusMetricSender).Send(0xc00000e2d8, 0xc000027648, 0x8, 0xc000027666, 0x7, 0xc000025560, 0x1d, 0xc000152db0, 0xc000147620, 0x6, ...)
	/opcache-dashboard/metrics/prometheus_collector.go:69 +0x492
```

This line sends to prometheus next value:

```
    "apcu": {
        "enabled": true,
        "smaInfo": {
            "num_seg": 1,
            "seg_size": 209715080,
            "avail_mem": 193863696, <------------------------
            "block_lists": [
```

`smaInfo` may be null only if APCU cache disabled. If you don't use APCU, thant this is the answer.

**Comment by sokil** [comment]
Fixed since 0.5.2

**Comment by allanian** [comment]
thx you!
